src/app/components/itemlist.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import useFetchAdvocates, { Advocate } from "../hooks/useFetchAdvocates";
+
+interface ItemListProps {
+	searchText: string | null;
+}
+
+const ItemList: React.FC<ItemListProps> = ({ searchText }: ItemListProps) => {
+
+	const [currentPage, setCurrentPage] = useState(1);
+	const [pageSize,] = useState(10);
+	const offset = (currentPage - 1) * pageSize;
+
+	const { advocates, resultCount, loading } = useFetchAdvocates("/api/advocates", pageSize, offset);
+
+	const totalPages = Math.ceil((resultCount ?? 0) / pageSize);
+
+	const nextPage = () => {
+		if (currentPage < totalPages) {
+			setCurrentPage(currentPage + 1);
+		}
+	};
+
+	const prevPage = () => {
+		if (currentPage > 1) {
+			setCurrentPage(currentPage - 1);
+		}
+	};
+
+	const filteredItems = advocates && advocates.filter(
+		(advocate: Advocate) => {
+			return (
+				advocate.firstName.toLowerCase().includes((searchText ?? "").toLocaleLowerCase()) ||
+				advocate.lastName.toLocaleLowerCase().includes((searchText ?? "").toLocaleLowerCase()) ||
+				advocate.city.toLocaleLowerCase().includes((searchText ?? "").toLocaleLowerCase()) ||
+				advocate.degree.toLocaleLowerCase().includes((searchText ?? "").toLocaleLowerCase()) ||
+				advocate.specialties.filter(str => new RegExp(searchText!, "i").test(str)).length > 0 ||
+				advocate.phoneNumber.toString() === searchText ||
+				advocate.yearsOfExperience.toString() === searchText
+			)
+		}
+	) || [];
+
+	return (
+
+		<React.Fragment>
+			<div className="grid grid-cols-7 border border-gray-300 ml-2 mr-2">
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">First Name</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">Last Name</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">City</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">Degree</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">Specialities</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">Years of Experience</div>
+				<div className="font-bold bg-gray-200 p-2 flex justify-center items-center">Phone Number</div>
+				<React.Fragment>
+					{!loading ?
+						(filteredItems.map((advocate: Advocate, index: number) => {
+							return (
+								<React.Fragment key={index}>
```

Comment:
I could have used a table to display data but I was feeling kind of daring and wanted to try a Flex/Grid based approach. Thinking about it now a React version similar to Vuetify's v-data-table component would be a better long term play. 

https://vuetifyjs.com/en/components/data-tables/basics/#usage